src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java

```diff
@@ -0,0 +1,24 @@
+package com.hubspot.jinjava.tree.parse;
+
+public class StrictWhitespaceControlParser implements WhitespaceControlParser {
+
+  @Override
+  public boolean hasLeftTrim(String unwrapped) {
+    return unwrapped.startsWith("-");
```

Comment:
Might as well have `unwrapped.charAt(0) == '-'`, which is faster

Reply:
Good call, given that this gets called a bunch throughout a whole template.

Reply:
I also added a unit test just to make sure empty expressions don't reach this point without any characters in them (so there's no need to check that the string isn't empty before indexing into it).